client/src/partials/purchase/view/purchase_records.js

```diff
@@ -1,24 +1,21 @@
-angular.module('bhima.controllers')
-.controller('purchaseRecords', [
-  '$scope',
-  '$q',
-  'connect',
-  function ($scope, $q, connect) {
-    var session = $scope.session = { purchase_type: 'indirect'};
-    $scope.purchase_filter = {};
+
+var purchaseRecords = function ($q, connect) {
```

Comment:
It is better to always have `angular.module()` as the first line in the controller.  [You can see an example here](https://github.com/johnpapa/angular-styleguide#separate-data-calls).  That way, you can easily find where Angular binds the controller and see what its name is.   Like this:
```js
angular.module('bhima.controllers')
.controller('PurchaseRecordsController', purchaseRecords);

purchaseRecords.$inject = [/* some injects */];

// this will be hoisted to the top of the controller by javascript!
function purchaseRecords(/* some names for the injects */) {
  // code ...
}
```